cmd/helm/root.go

```diff
@@ -208,5 +210,54 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 	// Check permissions on critical files
 	checkPerms()
 
+	// Check for expired repositories
+	checkForExpiredRepos(settings.RepositoryConfig)
+
 	return cmd, nil
 }
+
+func checkForExpiredRepos(repofile string) {
+
+	expiredRepos := []struct {
+		name string
+		old  string
+		new  string
+	}{
+		{
+			name: "stable",
+			old:  "kubernetes-charts.storage.googleapis.com",
+			new:  "https://charts.helm.sh/stable",
+		},
+		{
+			name: "incubator",
+			old:  "kubernetes-charts-incubator.storage.googleapis.com",
+			new:  "https://charts.helm.sh/incubator",
+		},
+	}
+
+	// parse repo file.
+	// Ignore the error because it is okay for a repo file to be unparseable at this
+	// stage. Later checks will trap the error and respond accordingly.
+	repoFile, err := repo.LoadFile(repofile)
```

Comment:
Do we really want to hit the disk and read this file on every command every time? I'm just thinking out loud. It is a small file but on many commands it will be read more than once.

Reply:
I'm open to alternatives...

Reply:
Valid observation from @mattfarina. Can't think of an alternative at moment.

Reply:
Currently, this is blocking the PR merge, but nobody has any alternatives. Not sure what you want me to do.